src/s2/s2region_term_indexer_test.cc

```diff
@@ -112,62 +116,80 @@ void TestRandomCaps(const S2RegionTermIndexer::Options& options,
-  options.set_min_level(0);                    // Use face cells.
-  options.set_max_level(16);
-  options.set_max_cells(20);
-  TestRandomCaps(options, QueryType::CAP);
+  DataType index_type{};
+  DataType query_type{};
+};
+
+// We run one test case for each combination:
+// index_type: POINT, CAP
+// query_type: POINT, CAP
+// optimize_for_space: false, true
+// index_contains_points_only: false, true
+// query_contains_points_only: false, true
+INSTANTIATE_TEST_CASE_P(
+    S2RegionTermIndexerTests, S2RegionTermIndexerTest,
+    testing::Combine(testing::Values(DataType::POINT, DataType::CAP),
+                     testing::Values(DataType::POINT, DataType::CAP),
+                     testing::Bool(), testing::Bool(), testing::Bool()),
+    [](const testing::TestParamInfo<TestCase>& info) {
+      return absl::StrCat(
+          "Index", std::get<0>(info.param), "Query", std::get<1>(info.param),
+          "SpaceOpt", std::get<2>(info.param), "IndexOpt",
+          std::get<3>(info.param), "QueryOpt", std::get<4>(info.param));
+    });
+
+TEST_P(S2RegionTermIndexerTest, DefaultParametersValues) {
+  TestRandomCaps(options, index_type, query_type);
 }
 
-TEST(S2RegionTermIndexer, IndexRegionsQueryPointsOptimizeTime) {
-  S2RegionTermIndexer::Options options;
-  options.set_optimize_for_space(false);       // Optimize for time.
-  options.set_min_level(0);                    // Use face cells.
+TEST_P(S2RegionTermIndexerTest, UseFaceCells) {
+  options.set_min_level(0);
   options.set_max_level(16);
   options.set_max_cells(20);
-  TestRandomCaps(options, QueryType::POINT);
+  TestRandomCaps(options, index_type, query_type);
 }
 
-TEST(S2RegionTermIndexer, IndexRegionsQueryRegionsOptimizeTimeWithLevelMod) {
-  S2RegionTermIndexer::Options options;
-  options.set_optimize_for_space(false);       // Optimize for time.
-  options.set_min_level(6);                    // Constrain min/max levels.
+TEST_P(S2RegionTermIndexerTest, ConstrainMinMaxLevels) {
+  options.set_min_level(6);
   options.set_max_level(12);
   options.set_level_mod(3);
-  TestRandomCaps(options, QueryType::CAP);
+  TestRandomCaps(options, index_type, query_type);
 }
 
-TEST(S2RegionTermIndexer, IndexRegionsQueryRegionsOptimizeSpace) {
-  S2RegionTermIndexer::Options options;
-  options.set_optimize_for_space(true);        // Optimize for space.
+TEST_P(S2RegionTermIndexerTest, UseLeafCells) {
   options.set_min_level(4);
-  options.set_max_level(S2CellId::kMaxLevel);  // Use leaf cells.
+  options.set_max_level(S2CellId::kMaxLevel);
   options.set_max_cells(8);
-  TestRandomCaps(options, QueryType::CAP);
+  TestRandomCaps(options, index_type, query_type);
 }
 
-TEST(S2RegionTermIndexer, IndexPointsQueryRegionsOptimizeTime) {
-  S2RegionTermIndexer::Options options;
-  options.set_optimize_for_space(false);       // Optimize for time.
-  options.set_min_level(0);                    // Use face cells.
+TEST_P(S2RegionTermIndexerTest, UseFaceCells2) {
```

Comment:
Use something more descriptive than "2" to differentiate it.

Reply:
>Use something more descriptive than "2" to differentiate it.

What? No good name was in old code

Reply:
I can use UseFaceCellsLevelMode2?

Reply:
Add some more descriptive